textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/Dictionary.java

```diff
@@ -0,0 +1,33 @@
+package edu.uci.ics.textdb.dataflow.dictionarymatcher;
+
+import java.util.List;
+
+import edu.uci.ics.textdb.api.common.IDictionary;
+
+/**
+ * @author Sudeep [inkudo]
+ *
+ */
+public class Dictionary implements IDictionary {
+
+    private int cursor = -1;
+    private List<String> dict;
+
+    public Dictionary(List<String> dict) {
```

Comment:
It's confusing to pass a parameter called "dict" to a constructor of the class "Dictionary".  I suggest we make the type of the parameters more explicit, which is "stringList".  Make necessary changes to other places, e.g., member variable "dict".